src/SFA.DAS.Commitments.Database/StoredProcedures/UpdateCommitment.sql

```diff
@@ -9,14 +9,22 @@
 	@lastUpdatedByProviderName NVARCHAR(255),
 	@lastUpdatedByProviderEmail NVARCHAR(255)
 AS
+BEGIN
 
-UPDATE [dbo].[Commitment] 
-	SET CommitmentStatus = @commitmentStatus, 
-		EditStatus = @editStatus,
-		LastAction = @lastAction,
-		TransferApprovalStatus = @transferApprovalStatus,
-		LastUpdatedByEmployerName = @lastUpdatedByEmployerName, 
-		LastUpdatedByEmployerEmail = @lastUpdatedByEmployerEmail,
-		LastUpdatedByProviderName = @lastUpdatedByProviderName,
-		LastUpdatedByProviderEmail = @lastUpdatedByProviderEmail
-WHERE Id = @id;
+	UPDATE [dbo].[Commitment] 
+		SET CommitmentStatus = @commitmentStatus, 
+			EditStatus = @editStatus,
+			LastAction = @lastAction,
+			TransferApprovalStatus = @transferApprovalStatus,
+			LastUpdatedByEmployerName = @lastUpdatedByEmployerName, 
+			LastUpdatedByEmployerEmail = @lastUpdatedByEmployerEmail,
+			LastUpdatedByProviderName = @lastUpdatedByProviderName,
+			LastUpdatedByProviderEmail = @lastUpdatedByProviderEmail
+	WHERE Id = @id
+	AND EditStatus <> 0;
+
+
+	IF @@ROWCOUNT != 1 
```

Comment:
Would it be better to perform the update in a transaction and then rollback if @@ROWCOUNT is unexpected (before throwing the error)?